src/Analyzers/CSharp/Analyzers/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzer.cs

```diff
@@ -94,7 +94,7 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context)
             // if the target (the common expression in the pattern) is a method call,
             // then we can't guarantee that the rewritting won't have side-effects,
             // so we should warn the user
-            var isSafe = UnwrapImplicitConversion(pattern.Target) is not Operations.IInvocationOperation;
+            var isSafe = pattern.Target.UnwrapImplicitConversion() is not Operations.IInvocationOperation;
```

Comment:
this moved to be an extension (since around 5 different features had all copied the helper).